lib/pacemaker/pcmk_sched_primitive.c

```diff
@@ -1635,7 +1636,7 @@ pcmk__primitive_shutdown_lock(pcmk_resource_t *rsc)
 
     // Fence devices and remote connections can't be locked
     if (pcmk__str_eq(class, PCMK_RESOURCE_CLASS_STONITH, pcmk__str_null_matches)
```

Comment:
wonder if this can be `pcmk_is_set(rsc->flags, pcmk__rsc_fence_device)` like in `pcmk__primitive_internal_constraints()`

Reply:
definitely, but i'll leave that for a separate pr